### ISSUE TITLE
Unable to upgrade from 0.15.1 on FreeBSD

### ISSUE THREAD

**Issue by jeanpaulrh** [garethgeorge/backrest#550]
Note: if you have a question or need support please post in the [discussions area](https://github.com/garethgeorge/backrest/discussions).

**Describe the bug**
I tried today to upgrade from 0.15.1 in my FreeBSD Jail on Freenas. I unpacked the latest version and the logs say:

`{"level":"info","ts":1731318715.2121255,"msg":"decompressing bz2 archive (size=8905025)..."}
{"level":"info","ts":1731318719.5294273,"msg":"installed restic 0.17.1"}
{"level":"info","ts":1731318719.5309672,"msg":"migrating config from version 1 to 3"}
{"level":"fatal","ts":1731318719.5310442,"msg":"error loading config: config version 1 is too old to migrate, please try first upgrading to backrest 1.4.0 which is the last version that may be compatible with your config"}
`
Which is strange because I downloaded 0.16.1 which seems to be the latest release on github.
Then I tried to download 0.16.0 but I get the same error.
As a result my config.json get messed up and can't even start the old version (fortunately the script backs the configuration up so I have been able to restoer it).

**To Reproduce**
1. stop restic
2. unpack the new version
3. start the new version

**Expected behavior**

**Screenshots**

**Platform Info**

- FreeBSD (Freenas)

**Additional context**

**Comment by jeanpaulrh** [comment]
Another strange thing is that the webUI states 0.15.1 version.
When I start backrest though the logs say:
`2024-11-11T10:58:28.824+0100	INFO	Installing restic /home/gianpaolo/.local/share/backrest/restic-0.16.4 to 0.16.4...
2024-11-11T10:58:29.844+0100	INFO	Decompressing bz2 archive (size=8315170)...
2024-11-11T10:58:33.699+0100	INFO	Installed restic 0.16.4
`
which is odd.

**Comment by garethgeorge** [comment]
Hey, sorry you're running into this. Are you sure your earlier version was 0.15.1? Backrest has been running restic 0.17.0 or greater since https://github.com/garethgeorge/backrest/releases/tag/v1.4.0

There can be issues re: the browser caching the WebUI, you may need to refresh a few times or clear your cache.

**Comment by garethgeorge** [comment]
The number in the UI is the backrest version, the versions in the logs you linked though show backrest installing a version of restic, one that it last used in backrest version 1.4.0.

**Comment by jeanpaulrh** [comment]
ah ok. my bad. I should have read with more attention.
But, how is it possible that the web UI shows version 0.15.1 and the logs states that I am running an older version?
I would like to try to reinstall from scratch from 0.14 and the up but I'm afraid to lose my config.

**Comment by garethgeorge** [comment]
The steps I'd try following would be 

 1. Check that you only have one backrest install, and that it's in the correct location `/usr/local/bin/backrest` .
 2. Shutdown backrest, and remove the binary 
 3. Restore your last working config in `~/.config/backrest/config.json` (as you noticed backrest keeps backups so hopefully this is straightforward for you!).
 4. Install backrest 0.14.0 , let it run migrations.
 5. Install latest backrest.

That should bring you up to date -- sorry it's painful. I'll add that I don't expect to make significant config structure changes nearly as frequently in the future that would result in this sort of short lived migration (e.g. deprecated in a less than 6 mo time frame). Going forward 1 year of version compatibility is my target.

**Comment by garethgeorge** [comment]
To add one more note -- the UI version discrepancy is hard to explain. It makes me suspicious that somehow there are two instances running perhaps as different users on your system? It's possible that only one is successfully binding to the network?

**Comment by jeanpaulrh** [comment]
Sorry to have wasted your time.
I'm an idiot and I focused on the minor version not noticing I was still on 0.x
Moreover I forgot the server on which I was running the main instance of backrest and I was updating the wrong one.
Being it behind a proxy the real hostname was hidden and I realized only this morning what I was really doing. 
Sorry again and thanks for your software!
P.S. coming from 0.15 the new version is really awesome!!!
